subsys/debug/asan_hacks.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const char *__asan_default_options(void)
+{
+	return
+#if defined(CONFIG_64BIT) && defined(__GNUC__) && !defined(__clang__)
+		/* Running leak detection at exit could lead to a deadlock on
+		 * 64-bit boards if GCC is used.
+		 * https://github.com/zephyrproject-rtos/zephyr/issues/20122
+		 */
+		"leak_check_at_exit=0:"
+#endif
+		"";
+}
+
+const char *__lsan_default_suppressions(void)
+{
+	return
+#ifdef CONFIG_HAS_SDL
```

Comment:
Nit: Any reason to not just wrap the whole function with the ifdef? (I know this way it is more directly extensible, but it is also plugging into an interface we don't need to plug to otherwise)